packages/bun-types/bun.d.ts

```diff
@@ -5856,7 +5920,29 @@ declare module "bun" {
      * ```
      */
     cmds: string[],
-    options?: SpawnOptions.OptionsObject<In, Out, Err>,
+    options?: SpawnOptions.OptionsObject<In, Out, Err> & {
+      /**
+       * If true, stdout and stderr pipes will not automatically start reading
+       * data. Reading will only begin when you access the `stdout` or `stderr`
+       * properties.
+       *
+       * This can improve performance when you don't need to read output
+       * immediately.
+       *
+       * @default false
+       *
+       * @example
+       * ```ts
+       * const subprocess = Bun.spawn({
+       *   cmd: ["echo", "hello"],
+       *   lazy: true, // Don't start reading stdout until accessed
+       * });
+       * // stdout reading hasn't started yet
+       * await subprocess.stdout.text(); // Now reading starts
+       * ```
+       */
+      lazy?: boolean;
+    },
```

Comment:
I don't enjoy that this is duplicated -- Makes it difficult to maintain this docblock, since a reasonable engineer may forget to update both docblocks.

I would isolate this type out to be something like `SpawnOptions.OptionsObjectMaybeLazy<In, Out, Err>` or something like `SpawnOptions.OptionsObject<In, Out, Err> & SpawnOptions.SupportsLazyPipeOpts`.

The latter might be slightly nicer since you'll have to shoehorn `{ cmds: string[] }` too in the former `spawn` definition